src/client/__init__.py

```diff
@@ -2604,16 +2606,79 @@ def get_operation(self, operation_id: str) -> operations.Operation:
         response = self._request('get', f'/v1/operations/{operation_id}')
         return structure(response, operations.Operation)
 
+    @add_headers('client')
+    def wait_operation_with_progress(
+            self,
+            op: operations.Operation, timeout: datetime.timedelta = datetime.timedelta(minutes=10),
+    ) -> operations.Operation:
+        """Waits for the operation to complete, and return it. Showing progress bar while operation is in progress
+
+        Args:
+            op: ID of the operation.
+            timeout: How long to wait. Defaults to 10 minutes.
+
+        Raises:
+            TimeoutError: Raises it if the timeout has expired and the operation is still not completed.
+
+        Returns:
+            Operation: Completed operation.
+
+        Example:
+            Waiting for the pool to close can be running in the background.
+
+            >>> pool = toloka_client.get_pool(pool_id)
+            >>> while not pool.is_closed():
+            >>>     op = toloka_client.get_analytics([toloka.analytics_request.CompletionPercentagePoolAnalytics(subject_id=pool.id)])
+            >>>     op = toloka_client.wait_operation_with_progress(op)
+            >>>     percentage = op.details['value'][0]['result']['value']
+            >>>     print(
+            >>>         f'   {datetime.datetime.now().strftime("%H:%M:%S")}\t'
+            >>>         f'Pool {pool.id} - {percentage}%'
+            >>>         )
+            >>>     time.sleep(60 * minutes_to_wait)
+            >>>     pool = toloka_client.get_pool(pool.id)
+            >>> print('Pool was closed.')
+            ...
+        """
+        default_time_to_wait = datetime.timedelta(seconds=1)
+        default_initial_delay = datetime.timedelta(milliseconds=500)
+
+        if op.is_completed():
+            return op
+
+        utcnow = datetime.datetime.now(datetime.timezone.utc)
+        wait_until_time = utcnow + timeout
+
+        with logging_redirect_tqdm():
+            progress_bar = tqdm(total=100)
```

Comment:
Let's use `progress_bar` in the `with` statement as this will be [less error-prone](https://tqdm.github.io/docs/tqdm/#update).
```suggestion
            with tqdm(total=100) as progress_bar:
```